backend/wmg/data/load_cube.py

```diff
@@ -13,37 +13,186 @@
+
+    # TODO(prathap) - change to: return f"{data_root_path}/{data_schema_dir_path}/{file_name}"
+    # when working on: https://github.com/chanzuckerberg/single-cell-data-portal/issues/5166
+    return f"{data_root_path}/{file_name}"
+
+
+def _get_latest_snapshot_run_file_s3_key() -> str:
+    """
+    Return s3 key of latest_snapshot_run file
+    """
+    data_root_path = _get_wmg_s3_root_dir_path()
+    file_name = "latest_snapshot_run"
+
+    return f"{data_root_path}/{file_name}"
+
+
+def _get_wmg_s3_bucket_uri():
+    """
+    Get s3 bucket uri
+    """
+    return f"s3://{wmg_bucket_name}"
+
+
+def _get_wmg_s3_root_dir_path():
+    """
+    Get root directory in the s3 bucket under which the data artifact directories will be written to
+    """
+    return stack_name.strip("/") if stack_name else ""
 
-def _get_wmg_bucket_path():
-    if stack_name:
-        return f"s3://{wmg_bucket_name}{stack_name}"
-    else:
-        return f"s3://{wmg_bucket_name}"
+
+def _get_wmg_s3_data_schema_dir_path(snapshot_schema_version: str) -> str:
+    """
+    Get path for a particular data schema version under which data should be located
+    """
+    return f"snapshots/{snapshot_schema_version}"
+
+
+def _get_wmg_snapshot_s3_path(
+    snapshot_schema_version: str, snapshot_id: str, is_snapshot_validation_successful: bool
+) -> str:
+    """
+    Get the path of the directory in which the data artifact is located
+    """
+    snapshot_dir = snapshot_id if is_snapshot_validation_successful else f"{snapshot_id}_validation_failed_snapshot"
+
+    data_root_path = _get_wmg_s3_root_dir_path()
+    _get_wmg_s3_data_schema_dir_path(snapshot_schema_version)
+
+    # TODO(prathap) - change to: return f"{data_root_path}/{data_schema_dir_path}/{snapshot_dir}"
+    # when working on: https://github.com/chanzuckerberg/single-cell-data-portal/issues/5166
+    return f"{data_root_path}/{snapshot_dir}"
 
 
-def make_snapshot_active(snapshot_id):
+def _get_wmg_snapshot_s3_fullpath(
+    snapshot_schema_version: str, snapshot_id: str, is_snapshot_validation_successful: bool
+) -> str:
     """
-    Update snapshot pointer and remove older datasets
+    Get the fully qualified s3 location of the data artifact
     """
-    write_snapshot_id(snapshot_id)
-    remove_oldest_datasets(snapshot_id)
+    snapshot_path = _get_wmg_snapshot_s3_path(snapshot_schema_version, snapshot_id, is_snapshot_validation_successful)
 
+    return f"{_get_wmg_s3_bucket_uri()}/{snapshot_path}"
 
-def remove_oldest_datasets(timestamp):
+
+def _make_snapshot_active(snapshot_schema_version, snapshot_id):
     """
-    Remove all data stored under the oldest timestamp
+    Update the latest snapshot reference file to point to the newly generated snapshot and remove older snapshots
+    """
+    _write_latest_snapshot_identifier_file(snapshot_schema_version, snapshot_id)
+
+    data_root_path = _get_wmg_s3_root_dir_path()
+    _get_wmg_s3_data_schema_dir_path(snapshot_schema_version)
```

Comment:
what is this call supposed to do?

Reply:
will fix with a refactor here by consolidating some code